rasa/nlu/selectors/embedding_response_selector.py

```diff
@@ -0,0 +1,480 @@
+                label_ranking = [
+                    {"name": self.inv_label_dict[label_idx], "confidence": score}
+                    for label_idx, score in ranking
+                ]
+
+        key_placeholder = (
+            self.response_type if self.response_type else DEFAULT_OPEN_UTTERANCE_TYPE
+        )
+        message.set(
+            "respond_{0}_response".format(key_placeholder), label, add_to_output=True
+        )
+        message.set(
+            "respond_{0}_response_ranking".format(key_placeholder),
+            label_ranking,
+            add_to_output=True,
+        )
+
+    # training data helpers:
+    @staticmethod
+    def _create_label_dict(
+        training_data: "TrainingData", attribute: Text = "intent"
+    ) -> Dict[Text, int]:
+        """Create intent dictionary"""
+
+        distinct_labels = set(
+            [
+                example.get(attribute)
+                for example in training_data.intent_examples
+                if example.get(attribute)
+            ]
+        )
+        return {response: idx for idx, response in enumerate(sorted(distinct_labels))}
+
+    @staticmethod
+    def _find_example_for_label(label, examples, label_type="intent"):
+        for ex in examples:
+            if ex.get(label_type) == label:
+                return ex
+
+    # @staticmethod
+    def _create_encoded_labels(
+        self,
+        label_dict: Dict[Text, int],
+        training_data: "TrainingData",
+        attribute: Text = "intent",
+        attribute_feats: Text = "intent_features",
+    ) -> np.ndarray:
+        """Create matrix with intents encoded in rows as bag of words.
+        """
+
+        encoded_all_labels = []
+
+        for key, idx in label_dict.items():
+            encoded_all_labels.insert(
+                idx,
+                self._find_example_for_label(
+                    key, training_data.intent_examples, attribute
+                ).get(attribute_feats),
+            )
+
+        return np.array(encoded_all_labels)
+
+    # noinspection PyPep8Naming
+    def _create_session_data(
+        self,
+        training_data: "TrainingData",
+        label_dict: Dict[Text, int],
+        attribute: Text = "intent",
+    ) -> "train_utils.SessionData":
+        """Prepare data for training"""
+
+        X = np.stack(
+            [
+                e.get("text_features")
+                for e in training_data.intent_examples
+                if e.get(attribute)
+            ]
+        )
+
+        label_ids = np.array(
+            [
+                label_dict[e.get(attribute)]
+                for e in training_data.intent_examples
+                if e.get(attribute)
+            ]
+        )
+
+        Y = np.stack([self._encoded_all_label_ids[label] for label in label_ids])
+
+        return train_utils.SessionData(X=X, Y=Y, label_ids=label_ids)
+
+    def train(
+        self,
+        training_data: "TrainingData",
+        cfg: Optional["RasaNLUModelConfig"] = None,
```

Comment:
Please, try to not use abbreviations. It will be easier for others to understand the code if names are written out.
cfg -> config